src/clock.cr

```diff
@@ -0,0 +1,123 @@
+{% if flag?(:darwin) %}
+  require "c/mach/mach_time"
+{% end %}
+
+# Monotonic Clock.
+#
+# A monotonic clock is tied to the system and isn't affected by time
+# fluctuations, for example leap seconds or manually changing the computer's
+# time.
+#
+# Compared to `Time`, the `Clock` value isn't interesting and most likely means
+# nothing, but its value is guaranteed to be linearily increasing from a fixed
+# point in time (whose origin is unspecified), thus useful to calculate the
+# elapsed time between two clocks.
+#
+# Example:
+#
+# ```
+# timeout = 5.seconds
+# clock = Clock.monotonic
+#
+# until clock.elapsed?(timeout)
+#   do_domething
+# end
+# ```
+struct Clock
+  include Comparable(self)
+  include Comparable(Time::Span)
+  include Comparable(Float64 | Float32)
+
+  # :nodoc:
+  TicksPerSeconds = 1_000_000_000
```

Comment:
I realised, just wasn't sure if there was any need for the difference. We throw away precision in `Time.compute_second_and_tenth_microsecond` on linux. I think it's probably for the best to keep the higher precision here however.